csrc/torch_memory_saver.cpp

```diff
@@ -208,22 +248,36 @@ class TorchMemorySaver {
                 continue;
             }
 
-            CUmemGenericAllocationHandle newAllocHandle;
-            CUDAUtils::cu_mem_create(&newAllocHandle, metadata.size, metadata.device);
+            if (metadata.state == AllocationState::PAUSED) {
```

Comment:
nit: for symmetry, it seems that when we pause a paused memory we error, then what about we also error when we resume a resumed memory